test/Microsoft.ML.Functional.Tests/ModelFiles.cs

```diff
@@ -30,6 +32,84 @@ private class InputData
             public float[] Features { get; set; }
         }
 
+        /// <summary>
+        /// Model Files: The (minimum) nuget version can be found in the model file.
+        /// </summary>
+        [Fact]
+        public void DetermineNugetVersionFromModel()
+        {
+            var modelFile = GetDataPath($"backcompat{Path.DirectorySeparatorChar}keep-model.zip");
+            var versionFileName = @"TrainingInfo\Version.txt"; // Must use '\' for cross-platform testing.
+            using (ZipArchive archive = ZipFile.OpenRead(modelFile))
+            {
+                // The version of the entire model is kept in the version file.
+                var versionPath = archive.Entries.First(x => x.FullName == versionFileName);
+                Assert.NotNull(versionPath);
+                using (var stream = versionPath.Open())
+                using (var reader = new StreamReader(stream))
+                {
+                    // The only line in the file is the version of the model.
+                    var line = reader.ReadLine();
+                    Assert.Equal(@"1.0.0.0", line);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Model Files: Save a model, including all transforms, then load and make predictions.
+        /// </summary>
+        /// <remarks>
+        /// Serves two scenarios:
+        ///  1. I can train a model and save it to a file, including transforms.
+        ///  2. Training and prediction happen in different processes (or even different machines). 
+        ///     The actual test will not run in different processes, but will simulate the idea that the 
+        ///     "communication pipe" is just a serialized model of some form.
```

Comment:
not necessary for tests files, but as FYI lists need to be in xml style as well.